packages/backend/test/peripherals/coingecko/CoingeckoQueryService.test.ts

```diff
@@ -415,3 +519,98 @@ describe(getFullTimestampsList.name, () => {
     })
   })
 })
+
+describe(generateRangesToCallHourly.name, () => {
+  it('30 days', () => {
+    const start = UnixTime.fromDate(new Date('2021-07-01T00:00:00Z'))
+
+    expect(generateRangesToCallHourly(start, start.add(30, 'days'))).toEqual([
+      {
+        start: start,
+        end: start.add(30, 'days'),
+      },
+    ])
+  })
+
+  it('90 days', () => {
+    const start = UnixTime.fromDate(new Date('2021-07-01T00:00:00Z'))
+
+    expect(generateRangesToCallHourly(start, start.add(90, 'days'))).toEqual([
+      {
+        start: start,
+        end: start.add(COINGECKO_HOURLY_MAX_SPAN_IN_DAYS, 'days'),
+      },
+      {
+        start: start.add(COINGECKO_HOURLY_MAX_SPAN_IN_DAYS, 'days'),
+        end: start.add(90, 'days'),
+      },
+    ])
+  })
+})
+
+describe.skip(CoingeckoQueryService.name + ' e2e tests', function () {
+  this.timeout(100000)
+
+  const COIN = CoingeckoId('ethereum')
+  const START = UnixTime.fromDate(new Date('2021-01-01T00:00:00Z'))
+  const DAYS_SPAN = 90
+  const MAX_FAULT_MINUTES = 25
+  const EXPECTED_HOURLY_FAULT_RATIO = 0.15
+
+  const coingeckoQueryService = new CoingeckoQueryService(
+    new CoingeckoClient(new HttpClient())
```

Comment:
Please create dependencies as separate variables as this is how it's done everywhere else.